### PR TITLE
Send a SIGTERM signal to the secret backend before SIGKILL

### PR DESCRIPTION
**Pull request by L3n41c**
### What does this PR do?

Send a `SIGTERM` signal to the secret backend before `SIGKILL`.

### Motivation

Now, if the secret backend is too slow to decode the secrets, a timeout fires in the `datadog-agent` that kills the secret backend with `SIGKILL`.
The secret backend cannot catch this signal to do some cleanup and log where is was blocked.

The goal of this change is to have the `datadog-agent` send a `SIGTERM` signal that the secret backend can catch before sending the `SIGKILL` signal that will forcibly terminate the secret backend.

This mimics the behaviour of `systemd`, `docker` or any other init system.

Combined with DataDog/datadog-vault-secrets#3, if the secret backend freezes, we’ll have a stack trace of where the secret backend was blocked.
This will give us a clue as wether the issue was a DNS resolution issue, or a network connectivity issue, or a vault issue.

### Describe your test plan

Package `datadog-agent` with a version of the `datadog-vault-secrets` that contains DataDog/datadog-vault-secrets#3 and validate that, in case of time-out, get get the details about where the secret backend was blocked.

### CODE REVIEW

File: pkg/config/config_template.yaml
Comment (olivielpeau) [inline pkg/config/config_template.yaml:431]:
```suggestion
# secret_backend_term_timeout: 5
```

File: pkg/secrets/fetch_secret.go
Comment (olivielpeau) [inline pkg/secrets/fetch_secret.go:81]: the `exitCode` is used as a tag value for telemetry, I think it's useful to keep `timeout` as a possible value, any reason to remove this?

File: releasenotes/notes/send-SIGTERM-to-the-secret-backend-f0d12454ad45e013.yaml
Comment (olivielpeau) [inline releasenotes/notes/send-SIGTERM-to-the-secret-backend-f0d12454ad45e013.yaml:12]: nit: can you mention the 2 config options to control these timeouts?

**Review by olivielpeau — APPROVED** [review]
LGTM overall, the implementation looks great 👌 (very nice handling of the hierarchy of timeouts with contexts)

Left a few comments, would appreciate a review from @hush-hush as well, but once my comments are addressed no need for further approval from me 👍

File: pkg/secrets/fetch_secret.go
Comment (hush-hush) [inline pkg/secrets/fetch_secret.go:84]: This should be a error in my opinion.

File: pkg/secrets/secrets.go
Comment (hush-hush) [inline pkg/secrets/secrets.go:46]: I think we should check that termTimeout < killTimeout and log a warning if not. This should also be done in the `secret` command (see https://github.com/DataDog/datadog-agent/blob/master/pkg/secrets/info.go).

**Review by hush-hush — COMMENTED** [review]
A few comment. My main concern is Windows, have we tested the new behavior on it ?

File: pkg/secrets/fetch_secret_darwin.go
Comment (hush-hush) [inline pkg/secrets/fetch_secret_darwin.go:1]: Why duplicate the linux code for darwin ? We could just have a windows version and a non windows version.

**Review by hush-hush — CHANGES_REQUESTED** [review]

**Review by kayayarai — COMMENTED** [review]
no documentation impact

**Review by derekwbrown — APPROVED** [review]
code lgtm. 
I would also want to see some active testing on Windows before merging.

**Pull request closed without merging**
